spec.md

```diff
@@ -179,6 +212,32 @@ physically present in the final circuit. Refer to the description of the
 instance statement for details on how to instantiate a module
 ([@sec:instances]).
 
+### Public Modules
+
+A public module is a module that is marked with the `public`{.firrtl} keyword.
+This means that this is a module that is usable external to this circuit.  A
+public module thereby has a defined ABI---knowing the FIRRTL representation is
+sufficient to know its compiled representation.
+
+A public module has a number of restrictions:
+
+1. A public module may not be instantiated in the current circuit.
```

Comment:
would it be acceptable to instantiate a public aka root module *as an ext module (would need a new API)* in the current circuit? If not, why not?

Reply:
:exploding_head: good question / fun idea!